spec/datatables/volunteer_datatable_spec.rb

```diff
@@ -0,0 +1,430 @@
+require "rails_helper"
+
+RSpec.describe "VolunteerDatatable" do
+  let(:org) { create :casa_org }
+  let(:supervisors) { create_list :supervisor, 3, casa_org: org }
+  let(:assigned_volunteers) { Volunteer.joins(:supervisor) }
+  let(:subject) { described_class.new(org.volunteers, params).as_json }
+
+  let(:additional_filters) do
+    {
+      active: %w[false true],
+      supervisor: supervisors.map(&:display_name),
+      transition_aged_youth: %w[false true]
+    }
+  end
+  let(:order_by) { "display_name" }
+  let(:order_direction) { "asc" }
+  let(:page) { 1 }
+  let(:per_page) { 10 }
+  let(:search_term) { nil }
+  let(:params) do
+    datatable_params(
+      additional_filters: additional_filters,
+      order_by: order_by,
+      order_direction: order_direction,
+      page: page,
+      per_page: per_page,
+      search_term: search_term
+    )
+  end
+
+  before do
```

Comment:
this runs before each test. Since we're only testing filtering not editing, maybe we can run this only once for this file instead of once per test in the file. https://relishapp.com/rspec/rspec-core/v/2-2/docs/hooks/before-and-after-hooks `before(:all)`